app/scripts/directives/service-definition-edit.js

```diff
@@ -80,7 +83,25 @@
         scope.transformToYamlDocumentFragment = function (editedJson) {
           var yamlFrag = {};
           angular.forEach(editedJson, function(svalue) {
-            yamlFrag[svalue.name] = svalue.value;
+            if (svalue.name === 'environment') {
+              var obj = {};
+              angular.forEach(svalue.value, function (sv) {
+                if (sv !== '') {
+                  var hash = [];
+                  // if incoming value has = syntax
+                  if (sv.indexOf('=') !== -1) {
+                    hash = sv.split('=');
+                  } else {
+                    // if incoming value has : syntax
+                    hash = sv.split(':');
+                  }
```

Comment:
@dharmamike I was hoping I would just pick up the first ':' or '='. Wonder how docer-compose would parse it? Let me test it out.